ovs/dal/datalist.py

```diff
@@ -203,7 +203,13 @@ def _execute_query(self):
         prefix = '{0}_{1}_'.format(DataObject.NAMESPACE, object_type_name)
 
         if self._guids is not None:
-            entries = list(self._persistent.get_multi(['{0}{1}'.format(prefix, guid) for guid in self._guids]))
+            entries = []
+            for guid in self._guids[:]:
+                try:
+                    entries.append(self._persistent.get('{0}{1}'.format(prefix, guid)))
```

Comment:
That will introduce an enormous slowdown of the code in case of a missing key. I think a better approach might be to catch the ```KeyNotFoundException``` and remove the reported field from the list, after which you can execute a multi-get again. Maybe you can for example do a maximum of 5 tries (we also retry 5 times in other places when we suffer from race conditions).

It's probably a good idea to first create a separate guid_list or sth like that, so you can easily remove entries from it.